config/create_fortran_test.py

```diff
@@ -0,0 +1,138 @@
+#
+# For bob example:
+#
+# $ python3 config/create_fortran_test.py -j ../../../exercism/problem-specifications/exercises/bob/canonical-data.json -t exercises/bob/bob_test.f90
+# Namespace(json='../../../exercism/problem-specifications/exercises/bob/canonical-data.json', target='exercises/bob/bob_test.f90')
+# Wrote : exercises/bob/bob_test.f90
+# $ cp config/CMakeLists.txt exercises/bob/.
+# $ cd exercises/bob
+# $ touch bob.f90
+# $ mkdir Debug
+# $ cd Debug
+# $ cmake ..
+# $ make 
+# $ ctest -V
+
+
+import json
+import argparse
+
+def fix_and_quote_fortran_multiline(txt):
+    if isinstance(txt, str):
+        txt=txt.replace('\n', '"// & \n    & "')
+        return '"%s"'%txt
+    else:
+        return txt
+
+def write_testcase(c, tnum):    
+    si=[]
+        # {'description': 'non-question ending with whitespace', 'property': 'response', 'input': {'heyBob': 'This is a statement ending with whitespace      '}, 'expected': 'Whatever.'}
+    description = c['description']
+    fcall = c['property']
+    args = [v for k,v in c['input'].items()]
+    inp = '{}({}'.format( fcall, fix_and_quote_fortran_multiline(args[0]))
+    for a in args[1:]:
+        inp='{}, {}'.format(inp, fix_and_quote_fortran_multiline(a))
+    inp='{})'.format(inp)
+    expected = c['expected']        
+    expected = fix_and_quote_fortran_multiline(expected) 
+    si.append('  ! Test %d: %s'%(tnum+1, description))        
+    si.append('  call assert_equal({}, {}, "{}")'.format(expected, inp, description))
+    return si
+
+
+def create_single_test(j):
+    si=[]
+    for tnum,c in enumerate(j['cases']):
+        # {'description': 'non-question ending with whitespace', 'property': 'response', 'input': {'heyBob': 'This is a statement ending with whitespace      '}, 'expected': 'Whatever.'}
+        if 'cases' in c:
+            si.extend(create_single_test(c))
+        else:
+            si.extend(write_testcase(c, tnum))
+    return si
+
+    
+
+def create_test(test_name, json_name):
+    f = open (json_name)
+    j=json.load(f)
+    f.close()
+    header="""
+! This test was created from JSON    
+! version: EXERCISE_VERSION    
+!
+
+"""
+    program="""
+    
+program EXERCISE_test_main
+  use TesterMain
+  use EXERCISE
+  implicit none
+
+    """
+
+    exercise_version = j['version']
+    exercise = j['exercise']
+    exercise = exercise.replace('-','_')
+
+    s=header.replace('JSON', json_name).replace('EXERCISE_VERSION',exercise_version)    
+    s=s+program.replace('EXERCISE', exercise )
+    si=s.splitlines()
+    si.extend(create_single_test(j))
+
+    si.append(' ')
+    si.append('  call test_end()')
+    si.append(' ')
+    si.append('end program')
+    si.append(' ')
+    #json.dumps(j)
+    of=open(test_name, 'w')
+    for ss in si:
+        of.write('%s\n'%ss)
+    of.close()
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(description='Create test.f90')
+    
+    parser.add_argument("-j", "--json",
```

Comment:
This is personal preference: Imo positional args instead of flags would be nicer here because both arguments are required all the time.

Reply:
Do you mean to say that it would be nicer here "otherwise" both arguments are required...? 

 Is there a way that one can give the value for the other if given easily?  So both forms are available, long form and short form?

Reply:
Both arguments are marked as `required=True,`. I was trying to say that I would prefer it if one could use `python3 config/create_fortran_test.py ../../../exercism/problem-specifications/exercises/bob/canonical-data.json exercises/bob/bob_test.f90` instead of `python3 config/create_fortran_test.py -j ../../../exercism/problem-specifications/exercises/bob/canonical-data.json -t exercises/bob/bob_test.f90`.

Though after checking if there are any standards or conventions for when to use positional arguments over option-based arguments (`--foo`), it seems that target/output files [should always use options](https://www.gnu.org/prep/standards/standards.html#Command_002dLine-Interfaces). Therefore I don't really mind if you follow that convention (input as positional/ordinary argument and output with a letter) or leave it as is.

Reply:
Sorry, but I really dont like positional arguments as soon as you want to extend the argument list. But you are right; for two simple arguments (in and out data) positional would be an option. But I would like to keep the argparse options because it is so easy to extend. I hope you are OK with this decision.

Reply:
Yea, I'm fine with it